### ISSUE TITLE
A Mastodon Widget to add a timeline/hashtag to a website

### ISSUE THREAD

**Issue by McPringle** [mastodon/mastodon#22954]
### Pitch

It is very easy to add a Twitter timeline to my own website. I would be a great benefit for many people running a website to easily embed a Mastodon widget on their website.

### Motivation

We are running the website https://hackergarten.net/ where we have a Twitter widget showing Hackergarten tweets. We would like to do the same with Mastodon. Actually it looks like we need to develop an own widget using the REST API of Mastodon. If every website owner need to develop their own widget, this simple task is is unnecessary complex. It would be much easier, if there is an official widget where you just need to specify the hashtag or profile name and it outputs the HTML you have to embed on your website.

**Comment by rcanada** [comment]
Have you tried using an RSS feed widget and the RSS feed of either the local timeline or a hashtag on the local timeline?

**Comment by McPringle** [comment]
Yes. Did not work because of cross site scripting issues. A script embedded in a website is only allowed to access resources from the same domain the script was loaded from.

**Comment by idotj** [comment]
Hi there, I have being working on a embed timeline for my website. The project is alive and got some users feedback to improve it. Have a look and feel free to open an issue if you see any changes needed :)
https://gitlab.com/idotj/mastodon-embed-feed-timeline
